### PR TITLE
MM-15343 Look in post metadata for OpenGraph image dimensions

### PR DESCRIPTION
**Pull request by hmhealey**
When the website doesn't provide dimensions for the image linked in it's OpenGraph metadata, the server calculates them, but it doesn't add those back to the OpenGraph metadata. The client needs to look at `post.metadata.images` to find them.

I couldn't resist doing some additional cleanup to that component and its utility functions, but the important part is the changes to `getBestImageUrl` so that it looks at `post.metadata.images` now.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-15343

### CODE REVIEW

File: components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx [hunk 1]
```diff
@@ -54,63 +55,44 @@ export default class PostAttachmentOpenGraph extends React.PureComponent {
...
+        const {height, width} = dimensions;
 
-    hasLargeImage({height, width}) {
-        let hasLargeImage;
         const largeImageMinRatio = 16 / 9;
         const largeImageMinWidth = 150;
 
         const imageRatio = width / height;
+
+        let hasLargeImage;
```
Comment (lieut-data) [inline components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx]: Worth just defining as the following?
```js
const hasLargeImage = width >= largeImageMinWidth && imageRatio >= largeImageMinRatio
```

File: components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx [hunk 2]
```diff
@@ -205,20 +166,17 @@ export default class PostAttachmentOpenGraph extends React.PureComponent {
...
+        return this.props.actions.editPost(patchedPost);
     }
 
-    isRemovePreview(post) {
-        if (post && post.props) {
-            return post.props[PostTypes.REMOVE_LINK_PREVIEW] && post.props[PostTypes.REMOVE_LINK_PREVIEW] === 'true';
+    hasPreviewBeenRemoved() {
+        const {post} = this.props;
+
+        if (!post || !post.props) {
```
Comment (lieut-data) [inline components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx]: Did we ever enable the setting to allow something like (can't recall the exact syntax):
```js
return post?.props?.[PostTypes.REMOVE_LINK_PREVIEW] === 'true'
```

File: components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx (hunk 1 above)
Comment (hmhealey) [reply components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx]: Oh yeah, that's much better

File: components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx (hunk 2 above)
Comment (hmhealey) [reply components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx]: I don't think we ever enabled that on the web app. That is the correct syntax though.

**Pull request merged**